olive/passes/pytorch/awq.py

```diff
@@ -0,0 +1,224 @@
+            ),
+            "duo_scaling": PassConfigParam(
+                type_=bool,
+                default_value=True,
+                description="Whether to scale using both w/x(True) or just x(False).",
+            ),
+            "modules_to_not_convert": PassConfigParam(
+                type_=list,
+                default_value=[],
+                description=(
+                    "The list of modules to not quantize, useful for quantizing models that explicitly "
+                    "require to have some modules left in their original precision (e.g. Whisper encoder, "
+                    "Llava encoder, Mixtral gate layers). Please refer to AutoAWQ documentation for "
+                    "quantizing HF models."
+                ),
+            ),
+            "export_compatible": PassConfigParam(
+                type_=bool,
+                default_value=False,
+                description=(
+                    "If True, this argument avoids real quantization by only applying "
+                    "the scales quantizing down to FP16."
+                ),
+            ),
+            "data_config": PassConfigParam(
+                type_=Union[DataConfig, Dict],
+                default_value=None,
+                description="Data config for quantization. Default value is None.",
+            ),
+            "pack_model_for_onnx_conversion": PassConfigParam(
+                type_=bool,
+                default_value=False,
+                description="Whether to pack the model for ONNX conversion. Default is False.",
+            ),
+        }
+
+    @torch.no_grad()
+    def _run_for_config(
+        self, model: PyTorchModelHandler, data_root: str, config: Dict[str, Any], output_model_path: str
+    ) -> PyTorchModelHandler:
+        from awq import AutoAWQForCausalLM
+        from awq.models import base as awq_model_base
+        from awq.quantize.quantizer import AwqQuantizer as AutoAwqQuantizer
+        from transformers import AutoTokenizer
+
+        if not torch.cuda.is_available():
+            raise ValueError("Please use GPU to run gptq quantization.")
+        elif self.host_device != Device.GPU:
+            logger.warning(
+                "GPTQ quantization requires GPU but the host device is %s, will ignore the host device",
+                self.host_device,
+            )
+
+        data_kwargs = {}
+        if config["data_config"]:
+            # set default values for data config
+            data_kwargs.update(
+                {
+                    "calib_data": config["data_config"].params_config.get("data_name"),
+                    "split": config["data_config"].params_config.get("split"),
+                    "text_column": config["data_config"].params_config.get("input_cols"),
+                }
+            )
+
+        quantizer = (
+            self._pack_model_for_onnx_conversion(config)
+            if config["pack_model_for_onnx_conversion"]
+            else AutoAwqQuantizer
```

Comment:
Please add a comment discussing how the packing is done differently to make it suitable for onnx conversion.

Reply:
Done with: 
```
pack_model_for_onnx_conversion is a flag to switch between the two quantizers
1. PyAutoAWQQuantizer is a quantizer implemented by autoawq package which is used by default
for quantizing the model. But it does not work with ONNX conversion since there are some
operations that are not supported by ONNX. So, we have to pack the model for ONNX conversion
2. That is why we have another quantizer method self._pack_model_for_onnx_conversion(config) to
return OrtAutoAWQQuantizer which is used for ONNX conversion.
```